core-web/apps/dotcms-ui/src/app/shared/dot-utils.ts

```diff
@@ -16,3 +18,15 @@ export function formatMessage(message: string, args: string[]): string {
         return typeof args[number] !== 'undefined' ? args[number] : match;
     });
 }
+
+// Generates an unique Url with host, language and device Ids
+export function generateDotFavoritePageUrl(pageState: DotPageRenderState): string {
+    return (
+        `${pageState.params.page?.pageURI}?` +
+        (pageState.params.site?.identifier ? `host_id=${pageState.params.site?.identifier}` : '') +
+        `&language_id=${pageState.params.viewAs.language.id}` +
+        (pageState.params.viewAs.device?.identifier
+            ? `&device_id=${pageState.params.viewAs.device?.identifier}`
+            : '')
+    );
+}
```

Comment:
How about doing something like that? Just to make it more readable.

```suggestion
export function generateDotFavoritePageUrl(pageState: DotPageRenderState): string {
  const { page, site, viewAs } = pageState.params;
    return (
        `${page?.pageURI}?` +
        (site?.identifier ? `host_id=${site?.identifier}` : '') +
        `&language_id=${viewAs.language.id}` +
        (viewAs.device?.identifier ? `&device_id=${viewAs.device?.identifier}` : '')
    );
}
```